Context.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+const EventEmitter = require('events');
+const IncomingWebSocket = require('./IncomingWebSocket');
+const OutgoingWebSocket = require('./OutgoingWebSocket');
+const logger = require('./loggerFactory');
+const { createTarget } = require('./util');
+const {
+  kSender,
+  kUpstreamClosed,
+  kStartConnection,
+  kMessageReceived,
+  kError,
+  kSendMessage,
+  kQueueMessage,
+  kDrainMessage,
+  kConnectionOpened,
+  kDequeueMessage,
+  kClientClosed,
+  kCleanup,
+  kDrainCompleted,
+  kReleaseTap,
+  kAddNewContext,
+  kUpstreamRestart,
+  config,
+  INCOMING,
+  OUTGOING
+} = require('./constants');
+
+/**
+ * Context holds the information for the incoming and outgoing sockets.
+ *
+ * Instantiates the incoming and outgoing sockets and registers the listeners.
+ * Acts as a bridge between incoming and outgoing sockets.
+ */
+class Context extends EventEmitter {
+
+  /**
+   * Creates the context with connection identifier.
+   *
+   * @param {string} connectionId
+   */
+  constructor(connectionId) {
+    super();
+    this.connectionId = connectionId;
+    this.incomingSocket = null;
+    this.outgoingSocket = null;
+    this.incomingLock = true;
+    this.outgoingLock = false;
```

Comment:
This should also be enabled since we are not sure if the outgoing websocket is created or not.

Reply:
checking... tried with true tap was not getting release properly hence, messages were getting queued.

Reply:
As discussed keeping as it is